### ISSUE TITLE
Check if file exists in repo

### ISSUE THREAD

**Issue by JanDevDE** [gmessner/gitlab4j-api#224]
Is there a good way to check if a file exists in a repository?

(e.g getRepositoryFileApi().getFile(X, Y ,Z).isExistent())

**Comment by gmessner** [comment]
@JanDevDE 
Currently RepositoryFileApi.getFile() fetches the file and its content. There is a way to just get the file meta-data (using HEAD), but this is not currently implemented in this library.

There will be a release later in the week, and this feature will be in that.  Will let you know when it is available.

**Comment by JanDevDE** [comment]
Great! :)

**Comment by gmessner** [comment]
@JanDevDE 
4.8.36 has been released with this functionality.  See the following in RepositoryFileApi:
```
public RepositoryFile getFileInfo(Object projectIdOrPath, String filePath, String ref)
```
Please take a look and close this issue if it meets your needs.

**Comment by JanDevDE** [comment]
Yep, that's cool. If the file is not in the gitlab repository, will it return a GitLabApiExcepion with Reason 404 ?

**Comment by gmessner** [comment]
@JanDevDE 
Yes,  ```getFileInfo()``` will throw a GitLabApiException if the file is not found.  The exception has a ```getHttpStatus()``` method to fetch the 404.

I just released 4.8.37 which includes the ```getOptionalFileInfo()``` method which does not throw an exception.  You use it as follows:
```
Optional<RepositoryFile> fileInfo = gitLabApi.getRepositoryFileApi().getOptionalFileInfo(project.getId(), "README", "master");
if (fileInfo.isPresent()) {
    System.out.println("file exists");
    RepositoryFile file = fileInfo.get();
    System.out.println("File size=" + file.getSize());
} else {
     System.out.println("Error getting file info, HTTP status=" +              
        GitLabApi.getOptionalException(fileInfo).getHttpStatus());
}
```

**Comment by JanDevDE** [comment]
Ah, thank you very much. I'll close this issue then. 👍
